pkg/cmd/pulumi/crypto_local.go

```diff
@@ -35,8 +37,27 @@ func readPassphrase(prompt string) (string, error) {
 	if phrase, ok := os.LookupEnv("PULUMI_CONFIG_PASSPHRASE"); ok {
 		return phrase, nil
 	}
+	if phraseFile, ok := os.LookupEnv("PULUMI_CONFIG_PASSPHRASE_FILE"); ok {
+		phraseFilePath, err := filepath.Abs(phraseFile)
+		if err != nil {
+			return "", errors.New("unable to construct a path the PULUMI_CONFIG_PASSPHRASE_FILE")
+		}
+		f, err := os.Open(phraseFilePath)
+		if err != nil {
+			return "", errors.New("unable to open PULUMI_CONFIG_PASSPHRASE_FILE")
+		}
+
+		reader := bufio.NewReader(f)
+		passphraseDetails, err := reader.ReadString('\n')
+		if err != nil {
+			return "", errors.New("unable to read PULUMI_CONFIG_PASSPHRASE_FILE")
+		}
+
+		return cmdutil.RemoveTrailingNewline(passphraseDetails), nil
```

Comment:
Sorry, looking this over now, prob easiest to read it as before, but just trim the string before returning it as Lee suggested.

Reply:
fixed!